src/pytorch_lightning/trainer/trainer.py

```diff
@@ -456,6 +456,65 @@ def __init__(
 
+        # Declare attributes to be set in _callback_connector on_trainer_init
+        self._callback_connector.on_trainer_init(
+            callbacks,
+            enable_checkpointing,
+            enable_progress_bar,
+            default_root_dir,
+            weights_save_path,
+            enable_model_summary,
+            max_time,
+            accumulate_grad_batches,
+        )
+
+        # init logger flags
+        self._loggers: List[Logger]
+        self._logger_connector.on_trainer_init(logger, log_every_n_steps, move_metrics_to_cpu)
+
+        # init debugging flags
+        if isinstance(fast_dev_run, int) and (fast_dev_run < 0):
+            raise MisconfigurationException(
+                f"fast_dev_run={fast_dev_run!r} is not a valid configuration. It should be >= 0."
+            )
+        self.fast_dev_run = fast_dev_run
+
+        # set fast_dev_run=True when it is 1, used while logging
+        if fast_dev_run == 1:
+            self.fast_dev_run = True
+
+        self.overfit_batches = _determine_batch_limits(overfit_batches, "overfit_batches")
+        overfit_batches_enabled = overfit_batches > 0
+
+        if fast_dev_run:
+            num_batches = int(fast_dev_run)
+            if not overfit_batches_enabled:
+                self.limit_train_batches = num_batches
+                self.limit_val_batches = num_batches
+            self.limit_test_batches = num_batches
+            self.limit_predict_batches = num_batches
+            max_steps = num_batches
+            self.num_sanity_val_steps = 0
+            max_epochs = 1
+            self.val_check_interval = 1.0
+            self.check_val_every_n_epoch = 1
+            self.loggers = [DummyLogger()] if self.loggers else []
+            rank_zero_info(
+                f"Running in `fast_dev_run` mode: will run the requested loop using {num_batches} batch(es). "
+                "Logging and checkpointing is suppressed."
+            )
+        else:
+            if not overfit_batches_enabled:
+                self.limit_train_batches = _determine_batch_limits(limit_train_batches, "limit_train_batches")
+                self.limit_val_batches = _determine_batch_limits(limit_val_batches, "limit_val_batches")
+            self.limit_test_batches = _determine_batch_limits(limit_test_batches, "limit_test_batches")
+            self.limit_predict_batches = _determine_batch_limits(limit_predict_batches, "limit_predict_batches")
+            self.num_sanity_val_steps = float("inf") if num_sanity_val_steps == -1 else num_sanity_val_steps
+            self.val_check_interval = _determine_batch_limits(val_check_interval, "val_check_interval")
+        if overfit_batches_enabled:
+            self.limit_train_batches = overfit_batches
+            self.limit_val_batches = overfit_batches
+
         min_steps, max_steps, min_epochs, max_epochs, max_time = _parse_loop_limits(
             min_steps, max_steps, min_epochs, max_epochs, max_time
         )
```

Comment:
While `_parse_loop_limits` doesn't do much with it, max_time by construction of the function call could be modified. Previously it was passed to `self._callback_connector.on_trainer_init` after these lines but now it has moved to before.

In other words, before we did the parsing first before using the values, now it is the other way around.

An alternative solution that avoids the re-ordering would have been to pass the `fast_dev_run` flag down to the `_parse_loop_limits` to be able to avoid emitting the warning. Was this considered? If yes, what was the blocker for not being able to do it this way?

Reply:
> An alternative solution that avoids the re-ordering would have been to pass the `fast_dev_run` flag down to the `_parse_loop_limits` to be able to avoid emitting the warning. Was this considered?

This was actually my initial solution. Aside from testing, the entire PR was a single commit: https://github.com/Lightning-AI/lightning/pull/13262/commits/97e73aab50f6192bae29fc5d419c989ded21a85b.

> If yes, what was the blocker for not being able to do it this way?

https://github.com/Lightning-AI/lightning/pull/13262#discussion_r897041894